src/Tasks/Microsoft.Common.CurrentVersion.targets

```diff
@@ -2518,7 +2518,7 @@ Copyright (C) Microsoft Corporation. All rights reserved.
     <SDKReferenceRegistryRoot Condition="'$(SDKReferenceRegistryRoot)' == ''">Software\Microsoft\Microsoft SDKs</SDKReferenceRegistryRoot>
     <SDKReferenceDirectoryRoot Condition="'$(SDKReferenceDirectoryRoot)' == ''">$(LocalAppData)\Microsoft SDKs;$(MSBuildProgramFiles32)\Microsoft SDKs</SDKReferenceDirectoryRoot>
     <!-- Manifest driven extension SDK locations -->
-    <SDKExtensionDirectoryRoot Condition="'$(SDKExtensionDirectoryRoot)' == '' and '$(SDKIdentifier)' != ''">$(MSBuildProgramFiles32)\Microsoft SDKs\Windows Kits\10;$(MSBuildProgramFiles32)\Windows Kits\10</SDKExtensionDirectoryRoot>
+    <SDKExtensionDirectoryRoot Condition="'$(SDKExtensionDirectoryRoot)' == '' and '$(SDKIdentifier)' != ''">$(MSBuildProgramFiles32)\Microsoft SDKs\Windows Kits\10;$(WindowsKitsDirectory)</SDKExtensionDirectoryRoot>
```

Comment:
Could you try to utilize https://learn.microsoft.com/visualstudio/msbuild/property-functions?view=vs-2022#msbuild-getregistryvaluefromview ? It will allow avoid WindowsKitsDirectory property creation
e.g.
https://github.com/dotnet/msbuild/blob/a9341111228592c25c43ecf6858e7b39e525b5a1/src/MSBuild/app.config#L145

also, we need to probe this key:
HKEY_LOCAL_MACHINE\SOFTWARE\Wow6432Node\Microsoft\Windows Kits\Installed Roots@KitsRoot10

Reply:
Is there a scenario where this stuff is supported on `dotnet build`? I don't expect it to be.